pkg/daemon/daemon.go

```diff
@@ -1103,10 +1103,10 @@ func (dn *Daemon) validateOnDiskState(currentConfig *mcfgv1.MachineConfig) bool
 	return true
 }
 
-// getRefDigest parses a Docker/OCI image reference and returns
+// GetRefDigest parses a Docker/OCI image reference and returns
 // its digest, or an error if the string fails to parse as
 // a "canonical" image reference with a digest.
-func getRefDigest(ref string) (string, error) {
+func GetRefDigest(ref string) (string, error) {
```

Comment:
for later, we might want to get these funcs out of pkg/daemon into an helper/rhcos utilities package

Reply:
I think where we should likely go is having a high level API exposed by the daemon that's just called by `cmd/machine-config-daemon/main.go`.

Reply:
uhm, code under `cmd` isn't something we might want to have as API. What's under `pkg/daemon` might be a best fit for something like that for sure

Reply:
Yeah, that's what I meant.

Reply:
I'm blind, you're right sorry for the extra noise